# This is run to avoid an edge case where the latest version of a (probably sub)
# dependency being yanked leads to pip failing to backtrack the dep.
# This happened with `cachecontrol==0.12.12 ; python_version >= "3.7" and python_version < "4.0"`.
name: Verify dependency locks

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

on:
  pull_request:
    branches:
      - master
    paths: ["requirements.txt", "dev-requirements.txt"]

jobs:
  verify-pr-dep-changes:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@8eb1f6a495037164bea451156472f35fdd6bafc0
        with:
          submodules: "true"

      - name: Set up Python 3.11
        uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d
        with:
          python-version: "3.11"

      - name: install prerequisites
        run: |
          python -m pip install --upgrade pip wheel
          python -m pip install -r ./dev-requirements.txt

      - name: Verify dependency locks
        run: python -m nox -s verify-locks
